akka-javasdk/src/main/scala/akka/javasdk/impl/HttpEndpointDescriptorFactory.scala

```diff
@@ -51,7 +51,19 @@ private[javasdk] object HttpEndpointDescriptorFactory {
       instanceFactory: HttpEndpointConstructionContext => Any): HttpEndpointDescriptor = {
     assert(Reflect.isRestEndpoint(endpointClass))
 
-    val mainPath = Option(endpointClass.getAnnotation(classOf[HttpEndpoint])).map(_.value()).filterNot(_.isEmpty)
+    // make sure it is predictable - always starts with slash and ends with slash, if undefined, always "/"
+    val mainPath = Option(endpointClass.getAnnotation(classOf[HttpEndpoint])).map(_.value()) match {
+      case None     => "/"
+      case Some("") => "/"
+      case Some(path) =>
+        val startingWithSlash =
+          if (!path.startsWith("/")) "/" + path
+          else path
```

Comment:
Minor suggestion, but easier to read without `!`.

```suggestion
          if (path.startsWith("/")) path
          else "/" + path
```

Also aligns with the next lines. 
The `else` branch changes the path by prepending or appending a `/`.